packages/amplify-category-geo/src/provider-utils/awscloudformation/index.ts

```diff
@@ -0,0 +1,228 @@
+import { convertToCompleteMapParams, isCompleteMapParams, MapParameters, ProviderContext } from './utils/mapParams';
+import { merge } from './utils/resourceUtils';
+import { supportedServices, ServiceConfig } from '../supportedServices';
+import { ServiceName, provider } from './utils/constants';
+import _ from 'lodash';
+import { open, exitOnNextTick } from 'amplify-cli-core';
+import { createMapResource, modifyMapResource, getCurrentMapParameters } from './utils/mapResourceUtils';
+import { $TSContext } from 'amplify-cli-core';
+import { removeWalkthrough } from '../../provider-utils/awscloudformation/service-walkthroughs/removeWalkthrough';
+import { category } from '../../constants';
+import { updateDefaultMapWalkthrough } from '../awscloudformation/service-walkthroughs/mapWalkthrough';
+
+/**
+ * Entry point for creating a new Geo resource
+ * @param context Amplify Core Context object
+ * @param service The cloud service that is providing the category
+ * @param parameters Parameters used to configure the resource. If not specified, a walkthrough will be launched to populate it.
+ */
+export async function addResource(
+  context: $TSContext,
+  service: string,
+  parameters?: Partial<MapParameters>
+): Promise<string> {
+  // load the service config for this service
+  const BAD_SERVICE_ERR = new Error(`amplify-category-geo is not configured to provide service type ${service}`);
+
+  while (!checkIfAuthExists(context)) {
```

Comment:
why does this need to be in a while loop? won't an `if` do?

Reply:
changed this to `if` with an error message asking to add `auth`